common/beerocks/bwl/include/bwl/mon_wlan_hal_types.h

```diff
@@ -348,6 +348,106 @@ struct SLinkMeasurementsResponse11k {
     uint32_t dmg_link_adapt_ack_reference_timestamp;
 };
 
+enum eChannelScanResultMode : uint8_t {
+    eMode_NA             = 0x0,
+    eMode_AdHoc          = 0x1,
+    eMode_Infrastructure = 0x2,
+};
+
+enum eChannelScanResultEncryptionMode : uint8_t {
+    eEncryption_Mode_NA   = 0x0,
+    eEncryption_Mode_AES  = 0x1,
+    eEncryption_Mode_TKIP = 0x2,
+};
+
+enum eChannelScanResultSecurityMode : uint8_t {
+    eSecurity_Mode_None = 0x0,
+    eSecurity_Mode_WEP  = 0x1,
+    eSecurity_Mode_WPA  = 0x2,
+    eSecurity_Mode_WPA2 = 0x3,
+};
+
+enum eChannelScanResultOperatingFrequencyBand : uint8_t {
+    eOperating_Freq_Band_NA     = 0x0,
+    eOperating_Freq_Band_2_4GHz = 0x1,
+    eOperating_Freq_Band_5GHz   = 0x2,
+};
+
+enum eChannelScanResultStandards : uint8_t {
+    eStandard_NA       = 0x0,
+    eStandard_802_11a  = 0x1,
+    eStandard_802_11b  = 0x2,
+    eStandard_802_11g  = 0x3,
+    eStandard_802_11n  = 0x4,
+    eStandard_802_11ac = 0x5,
+};
+
+enum eChannelScanResultChannelBandwidth : uint8_t {
+    eChannel_Bandwidth_NA     = 0x0,
+    eChannel_Bandwidth_20MHz  = 0x1,
+    eChannel_Bandwidth_40MHz  = 0x2,
+    eChannel_Bandwidth_80MHz  = 0x3,
+    eChannel_Bandwidth_160MHz = 0x4,
+    eChannel_Bandwidth_80_80  = 0x5,
+};
+
+typedef struct sChannelScanResults {
+    //The current service set identifier in use by the neighboring WiFi SSID. The value MAY be empty for hidden SSIDs.
+    char ssid[beerocks::message::WIFI_SSID_MAX_LENGTH];
+
+    //The BSSID used for the neighboring WiFi SSID.
+    sMacAddr bssid;
+
+    //The mode the neighboring WiFi radio is operating in. Enumerate
+    eChannelScanResultMode mode;
+
+    //The current radio channel used by the neighboring WiFi radio.
+    uint32_t channel;
+
+    //An indicator of radio signal strength (RSSI) of the neighboring WiFi radio measured in dBm, as an average of the last 100 packets received.
+    int32_t signal_strength_dBm;
+
+    //The type of encryption the neighboring WiFi SSID advertises. Enumerate List.
+    std::vector<eChannelScanResultSecurityMode> security_mode_enabled;
```

Comment:
I chose vector container because it suits the message parsing flow.
its goes like this:
in parse_info_elements() we go over each buffer message in a loop()
 in case  ie_type::TYPE_RSN is there we translate it to 
eChannelScanResultSecurityMode and push it to the "security_mode_enabled" vector.
then go back to the loop to see if there are more data in the buffer.

This applies to all vectors in the sChannelScanResults .

Reply:
The real question is whether there can be multiple security modes in the message...

Reply:
good question to the driver team.
I think that once we start using this data and not  just for statistics these kind of questions will become more important.

Reply:
Ok, so let's write it down and check later on with the driver team

Reply:
moved to #665